router_types.c

```diff
@@ -44,6 +47,22 @@ static const struct router_type *router_types[] = {
 	NULL,
 };
 
+void mac_filter_set(char *mac)
+{
+	uint8_t bytes[6];
+	int hex_values[6];
+	int i;
+	if (6 == sscanf(mac, "%x:%x:%x:%x:%x:%x%*c",
```

Comment:
Please fix this function and properly read the stuff. Something like:

    int read_mac(uint8_t mac[ETH_ALEN], const char *macstr)
    {
    	int ret;
    
    	ret = sscanf(macstr, "%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX",
    		     &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
    
    	if (ret != 6)
    		ret = sscanf(macstr, "%02hhX-%02hhX-%02hhX-%02hhX-%02hhX-%02hhX",
    			     &mac[0], &mac[1], &mac[2], &mac[3], &mac[4], &mac[5]);
    
    	return (ret == 6);
    
    }